File: Binary Tree Postorder Traversal/Go/solution1.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func Main() {}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
	result := make([]int, 0)

	if root == nil {
		return result
	}

	if root.Left != nil {
		arr := postorderTraversal(root.Left)
		result = append(result, arr...)
	}
	if root.Right != nil {
		arr := postorderTraversal(root.Right)
		result = append(result, arr...)
	}

	result = append(result, root.Val)

	return result
}

File: Two Sum IV - B-Tree/Python/solution1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        compMap = {}
        return self.helper(root, k, compMap)
        
    def helper(self, node, target, compMap):
        if node is None:
            return False
        # if the node val is a value we need for the target return true
        if node.val in compMap:
            return True
        # sub the node val from target to find the value we need to add to to reach the target
        # Add the comp to the map
        comp = target - node.val
        compMap[comp] = 1
        # recurse on the left and right branchs
        return self.helper(node.left, target, compMap) or \
                self.helper(node.right, target, compMap)

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right